### PR TITLE
Verify correct ProxiedUserDetails primaryUser in an all-server request

### PR DESCRIPTION
**Pull request by billoley**

### CODE REVIEW

**Review by keith-ratcliffe — APPROVED** [review]

**Review by ivakegg — APPROVED** [review]

File: src/main/java/datawave/microservice/authorization/user/ProxiedUserDetails.java [hunk 1]
```diff
@@ -24,33 +23,69 @@
 @XmlRootElement
 public class ProxiedUserDetails implements UserDetails {
     private String username;
-    private Set<DatawaveUser> proxiedUsers = new LinkedHashSet<>();
+    private List<DatawaveUser> proxiedUsers = new ArrayList<>();
```
Comment (jwomeara) [inline src/main/java/datawave/microservice/authorization/user/ProxiedUserDetails.java:26]: Would it ever be problematic to have duplicate proxied users in the list? Technically, we could have duplicate proxied users when using a list. A LinkedHashSet would give you control over the ordering, and handle deduplication (if needed).

File: src/main/java/datawave/microservice/authorization/user/ProxiedUserDetails.java [hunk 2]
```diff
@@ -24,33 +23,69 @@
...
+        List<DatawaveUser> users = new ArrayList<>();
+        users.add(primary);
+        if (datawaveUsers.size() > 1) {
+            // @formatter:off
+            // Skipping first user because if it is UserType.USER, it is the primary
+            // and already added.  If it is UserType.Server, then it will be added at the end
+            users.addAll(datawaveUsers.stream()
+                    .skip(1)
+                    .filter(u -> u != primary)
+                    .collect(Collectors.toList()));
```
Comment (jwomeara) [inline src/main/java/datawave/microservice/authorization/user/ProxiedUserDetails.java:81]: Brian used to add this comment for me all the time, so I'm going to add it for you...

Consider using forEach vs. creating a new list.

i.e.

```
datawaveUsers.stream()
   .skip(1)
   .filter(u -> u != primary)
   .forEach(x -> users.add(x));
```

**Review by jwomeara — CHANGES_REQUESTED** [review]

File: src/main/java/datawave/microservice/authorization/user/ProxiedUserDetails.java (hunk 2 above)
Comment (billoley) [reply src/main/java/datawave/microservice/authorization/user/ProxiedUserDetails.java]: @jwomeara - made changes

File: src/main/java/datawave/microservice/authorization/user/ProxiedUserDetails.java (hunk 1 above)
Comment (billoley) [reply src/main/java/datawave/microservice/authorization/user/ProxiedUserDetails.java]: @jwomeara - made changes

**Review by ivakegg — APPROVED** [review]

**Review by keith-ratcliffe — APPROVED** [review]

**Pull request merged**